sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -94,54 +96,74 @@ object JDBCRDD extends Logging {
     new StructType(columns.map(name => fieldMap(name)))
   }
 
-  /**
-   * Turns a single Filter into a String representing a SQL expression.
-   * Returns None for an unhandled filter.
-   */
-  def compileFilter(f: Filter, dialect: JdbcDialect): Option[String] = {
-
-    def quote(colName: String): String = {
-      val nameParts = SparkSession.active.sessionState.sqlParser.parseMultipartIdentifier(colName)
-      if(nameParts.length > 1) {
-        throw QueryCompilationErrors.commandNotSupportNestedColumnError("Filter push down", colName)
-      }
-      dialect.quoteIdentifier(nameParts.head)
+  private def checkColumnName(colName: String) = {
```

Comment:
The previous code is not just a check. It needs to go through the parser to strip quotes if exist

Reply:
Yes. But `V2ExpressionSQLBuilder` do the job too.

Reply:
It does not. V2 Predicate uses `String[]` to represent field reference, no need to quote. It's only a problem in v1 Filter and we must take care of it when converting v1 to v2.